tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs

```diff
@@ -129,14 +130,41 @@ internal RuntimeMetricsWriter(IDogStatsd statsd, TimeSpan delay, bool inAzureApp
 
         public void Dispose()
         {
+            if (Interlocked.CompareExchange(ref _disposed, 1, 0) == 1)
+            {
+                Log.Debug("Disposing Runtime Metrics but it was already disposed");
+                return;
+            }
+
+            Log.Debug("Disposing Runtime Metrics timer");
+            using (var manualResetEvent = new ManualResetEvent(false))
+            {
+                if (_timer.Dispose(manualResetEvent))
+                {
+                    if (!manualResetEvent.WaitOne(30000))
+                    {
+                        Log.Warning("Failed to dispose Runtime Metrics timer after 30 seconds");
+                    }
+                }
```

Comment:
nit: 30s is too long - we'll kill the process rather than wait that long, so I think 5s should be plenty... theoretically it's only the time it takes for a scheduled `PushEvents` loop to run one final time on the thread pool I think.. Also tiny nit for reducing nesting

```suggestion
                if (_timer.Dispose(manualResetEvent) &&!manualResetEvent.WaitOne(5_000))
                {
                    Log.Warning("Failed to dispose Runtime Metrics timer after 5 seconds");
                }
```